lib/AndroidVMHelper.cpp

```diff
@@ -0,0 +1,93 @@
+#include "AndroidVMHelper.h"
+
+static JavaVM *vmCache = nullptr;
+
+/// cached java classloader so that we can find our classes from other threads
+static jobject vcmiClassLoader;
+static jmethodID vcmiFindClassMethod;
+
+void AndroidVMHelper::cacheVM(JNIEnv *env)
+{
+    env->GetJavaVM(&vmCache);

```

Comment:
I think I'll fiiiinally have to configure my IDE to correctly format code for vcmi.